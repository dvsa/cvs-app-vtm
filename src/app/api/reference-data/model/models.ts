export * from './reference-data.model';
