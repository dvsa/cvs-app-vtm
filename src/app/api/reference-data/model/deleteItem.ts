/**
 * Reference Data Microservice
 * This is the API spec for the reference data service. The reference data will be stored in a AWS DynamoDB database.
 *
 * OpenAPI spec version: 1.0.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */

export interface DeleteItem { 
    success?: boolean;
}