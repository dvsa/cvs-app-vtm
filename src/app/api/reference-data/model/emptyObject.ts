/**
 * Reference Data Microservice
 * This is the API spec for the reference data service. The reference data will be stored in a AWS DynamoDB database.
 *
 * OpenAPI spec version: 1.0.0
 * Contact: test@test.com
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface EmptyObject {
}
