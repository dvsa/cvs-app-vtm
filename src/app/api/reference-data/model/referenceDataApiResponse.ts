/**
 * Reference Data Microservice
 * This is the API spec for the reference data service. The reference data will be stored in a AWS DynamoDB database.
 *
 * OpenAPI spec version: 1.0.0
 * Contact: test@test.com
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */
import { ReferenceDataApiResponseWithPagination } from './referenceDataApiResponseWithPagination';
import { ReferenceDataApiResponseWithoutPagination } from './referenceDataApiResponseWithoutPagination';

export type ReferenceDataApiResponse = ReferenceDataApiResponseWithoutPagination | ReferenceDataApiResponseWithPagination;