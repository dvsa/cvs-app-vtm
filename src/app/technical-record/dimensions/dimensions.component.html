<table class="govuk-table table-details">
  <thead class="govuk-table__head">
  <tr class="govuk-table__row">
    <th class="govuk-table__cell govuk-!-font-weight-bold">Length (mm)</th>
    <td id="test-length" class="govuk-table__cell">
      {{ activeRecord.dimensions.length | DefaultNullOrEmpty }}
    </td>
    <th class="govuk-table__cell"></th>
    <th class="govuk-table__cell"></th>
    <th class="govuk-table__cell"></th>
    <th class="govuk-table__cell"></th>
  </tr>
  </thead>
  <tbody class="govuk-table__body">
  <tr class="govuk-table__row">
    <th class="govuk-table__cell govuk-!-font-weight-bold">Width (mm)</th>
    <td id="test-width" class="govuk-table__cell">
      {{ activeRecord.dimensions.width | DefaultNullOrEmpty }}
    </td>
    <td class="govuk-table__cell"></td>
    <td class="govuk-table__cell"></td>
    <td class="govuk-table__cell"></td>
    <td class="govuk-table__cell"></td>
  </tr>
  <tr class="govuk-table__row" *ngIf="activeRecord.vehicleType == 'trl'">
    <th class="govuk-table__cell govuk-!-font-weight-bold">Front axle to rear axle</th>
    <td id="test-frontAxleToRearAxle" class="govuk-table__cell">
      {{ activeRecord.frontAxleToRearAxle | DefaultNullOrEmpty }}
    </td>
    <td class="govuk-table__cell"></td>
    <td class="govuk-table__cell"></td>
    <td class="govuk-table__cell"></td>
    <td class="govuk-table__cell"></td>
  </tr>
  <tr class="govuk-table__row" *ngIf="activeRecord.vehicleType == 'trl'">
    <th class="govuk-table__cell govuk-!-font-weight-bold">Rear axle to rear trailer</th>
    <td id="test-rearAxleToRearTrl" class="govuk-table__cell">
      {{ activeRecord.rearAxleToRearTrl | DefaultNullOrEmpty }}
    </td>
    <td class="govuk-table__cell"></td>
    <td class="govuk-table__cell"></td>
    <td class="govuk-table__cell"></td>
    <td class="govuk-table__cell"></td>
  </tr>
  <tr class="govuk-table__row">
    <th class="govuk-table__cell"></th>
    <td class="govuk-table__cell govuk-!-font-weight-bold">Minimum (mm)</td>
    <td class="govuk-table__cell govuk-!-font-weight-bold">Maximum (mm)</td>
    <td class="govuk-table__cell"></td>
    <td class="govuk-table__cell"></td>
    <td class="govuk-table__cell"></td>
  </tr>
  <tr class="govuk-table__row" *ngIf="activeRecord.vehicleType == 'hgv'">
    <th class="govuk-table__cell govuk-!-font-weight-bold"> Front of vehicle to 5th wheel coupling</th>
    <td id="test-frontAxleTo5thWheelCouplingMin" class="govuk-table__cell">
      {{ activeRecord.frontAxleTo5thWheelCouplingMin | DefaultNullOrEmpty }}
    </td>
    <td id="test-frontAxleTo5thWheelCouplingMax" class="govuk-table__cell">
      {{ activeRecord.frontAxleTo5thWheelCouplingMax | DefaultNullOrEmpty }}
    </td>
    <td class="govuk-table__cell"></td>
    <td class="govuk-table__cell"></td>
    <td class="govuk-table__cell"></td>
  </tr>
  <tr class="govuk-table__row" *ngIf="activeRecord.vehicleType == 'hgv'">
    <th class="govuk-table__cell govuk-!-font-weight-bold"> Front axle to 5th wheel</th>
    <td id="test-frontAxleTo5thWheelMin" class="govuk-table__cell">
      {{ activeRecord.frontAxleTo5thWheelMin | DefaultNullOrEmpty }}
    </td>
    <td id="test-frontAxleTo5thWheelMax" class="govuk-table__cell">
      {{ activeRecord.frontAxleTo5thWheelMax | DefaultNullOrEmpty }}
    </td>
    <td class="govuk-table__cell"></td>
    <td class="govuk-table__cell"></td>
    <td class="govuk-table__cell"></td>
  </tr>
  <tr class="govuk-table__row" *ngIf="activeRecord.vehicleType == 'trl'">
    <th class="govuk-table__cell govuk-!-font-weight-bold"> Coupling center to rear axle</th>
    <td id="test-couplingCenterToRearAxleMin" class="govuk-table__cell">
      {{ activeRecord.couplingCenterToRearAxleMin | DefaultNullOrEmpty }}
    </td>
    <td id="test-couplingCenterToRearAxleMax" class="govuk-table__cell">
      {{ activeRecord.couplingCenterToRearAxleMax | DefaultNullOrEmpty }}
    </td>
    <td class="govuk-table__cell"></td>
    <td class="govuk-table__cell"></td>
    <td class="govuk-table__cell"></td>
  </tr>
  <tr class="govuk-table__row" *ngIf="activeRecord.vehicleType == 'trl'">
    <th class="govuk-table__cell govuk-!-font-weight-bold">Coupling center to rear trailer</th>
    <td id="test-couplingCenterToRearTrlMin" class="govuk-table__cell">
      {{ activeRecord.couplingCenterToRearTrlMin | DefaultNullOrEmpty }}
    </td>
    <td id="test-couplingCenterToRearTrlMax" class="govuk-table__cell">
      {{ activeRecord.couplingCenterToRearTrlMax | DefaultNullOrEmpty }}
    </td>
    <td class="govuk-table__cell"></td>
    <td class="govuk-table__cell"></td>
    <td class="govuk-table__cell"></td>
  </tr>
  </tbody>
</table>
