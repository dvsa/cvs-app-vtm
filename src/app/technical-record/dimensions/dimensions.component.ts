import {Component, OnInit, ChangeDetectionStrategy, Input} from '@angular/core';
import {TechRecordHelpersService} from '@app/technical-record/tech-record-helpers.service';

@Component({
  selector: 'vtm-dimensions',
  templateUrl: './dimensions.component.html',
  styleUrls: ['../../app.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class DimensionsComponent implements OnInit {

  @Input() activeRecord: any;

  constructor(public techRecHelpers: TechRecordHelpersService) { }

  ngOnInit() {
  }

}
