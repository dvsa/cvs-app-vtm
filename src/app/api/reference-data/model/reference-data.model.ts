import { ReferenceDataResourceType } from '@models/reference-data.model';

export type ReferenceDataItem = {
  resourceType: ReferenceDataResourceType;
  resourceKey: string | number;
};

export type ReferenceDataList = Array<ReferenceDataItem>;

export type ReferenceDataApiResponse = {
  data: ReferenceDataList;
};
