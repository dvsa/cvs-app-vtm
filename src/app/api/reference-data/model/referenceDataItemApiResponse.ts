/**
 * Reference Data Microservice
 * This is the API spec for the reference data service. The reference data will be stored in a AWS DynamoDB database.
 *
 * OpenAPI spec version: 1.0.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */
import { EmptyObject } from './emptyObject';
import { ReferenceDataItem } from './referenceDataItem';

export type ReferenceDataItemApiResponse = ReferenceDataItem | EmptyObject;