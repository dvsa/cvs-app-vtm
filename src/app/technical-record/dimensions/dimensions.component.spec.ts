import {async, ComponentFixture, getTestBed, TestBed} from '@angular/core/testing';
import {SharedModule} from '@app/shared/shared.module';
import {CUSTOM_ELEMENTS_SCHEMA} from '@angular/core';

import { DimensionsComponent } from './dimensions.component';
import { TechRecord } from '@app/models/tech-record.model';

describe('DimensionsComponent', () => {

  let component: DimensionsComponent;
  let fixture: ComponentFixture<DimensionsComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [SharedModule],
      declarations: [
        DimensionsComponent,
      ],
      schemas: [CUSTOM_ELEMENTS_SCHEMA]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(DimensionsComponent);
    component = fixture.componentInstance;
    const tecRec: TechRecord = {
                                frontAxleTo5thWheelMin: 456,
                                frontAxleTo5thWheelMax: 660,
                                frontAxleTo5thWheelCouplingMin: 100,
                                frontAxleTo5thWheelCouplingMax: 150,
                                dimensions: {
                                            length: 100,
                                            width: 100,
                                            axleSpacing: [
                                              { axles: '1', value: 100},
                                              { axles: '2', value: 100},
                                              { axles: '3', value: 100},
                                            ]
                                            }
                              } as TechRecord;
    component.activeRecord = tecRec;
    component.axleSpacing = [
                              { axles: '1', value: 100},
                              { axles: '2', value: 100},
                              { axles: '3', value: 100},
                            ];
  });

  fit('should create view only with populated data', () => {
    fixture.detectChanges();

    expect(component).toBeDefined();
    expect(fixture).toMatchSnapshot();
  });

  afterAll(() => {
    TestBed.resetTestingModule();
  });

});
