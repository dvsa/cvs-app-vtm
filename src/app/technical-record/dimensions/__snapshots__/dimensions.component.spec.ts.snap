// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DimensionsComponent should create view only with populated data 1`] = `
<vtm-dimensions
  activeRecord={[Function Object]}
>
  <table
    class="govuk-table table-details"
  >
    <thead
      class="govuk-table__head"
    >
      <tr
        class="govuk-table__row"
      >
        <th
          class="govuk-table__cell govuk-!-font-weight-bold"
        >
          Length (mm)
        </th>
        <td
          class="govuk-table__cell"
          id="test-length"
        >
           100 
        </td>
        <th
          class="govuk-table__cell"
        />
        <th
          class="govuk-table__cell"
        />
        <th
          class="govuk-table__cell"
        />
        <th
          class="govuk-table__cell"
        />
      </tr>
    </thead>
    <tbody
      class="govuk-table__body"
    >
      <tr
        class="govuk-table__row"
      >
        <th
          class="govuk-table__cell govuk-!-font-weight-bold"
        >
          Width (mm)
        </th>
        <td
          class="govuk-table__cell"
          id="test-width"
        >
           100 
        </td>
        <td
          class="govuk-table__cell"
        />
        <td
          class="govuk-table__cell"
        />
        <td
          class="govuk-table__cell"
        />
        <td
          class="govuk-table__cell"
        />
      </tr>
      
      <tr
        class="govuk-table__row"
      >
        <th
          class="govuk-table__cell govuk-!-font-weight-bold"
        >
          Front axle to rear axle (mm)
        </th>
        <td
          class="govuk-table__cell"
          id="test-frontAxleToRearAxle"
        >
           - 
        </td>
        <td
          class="govuk-table__cell"
        />
        <td
          class="govuk-table__cell"
        />
        <td
          class="govuk-table__cell"
        />
        <td
          class="govuk-table__cell"
        />
      </tr>
      
      
      
      <tr
        class="govuk-table__row"
      >
        <th
          class="govuk-table__cell"
        />
        <td
          class="govuk-table__cell govuk-!-font-weight-bold"
        >
          Minimum (mm)
        </td>
        <td
          class="govuk-table__cell govuk-!-font-weight-bold"
        >
          Maximum (mm)
        </td>
        <td
          class="govuk-table__cell"
        />
        <td
          class="govuk-table__cell"
        />
        <td
          class="govuk-table__cell"
        />
      </tr>
      
      
      
      
    </tbody>
  </table>
</vtm-dimensions>
`;
